fastapi_pagination/ext/motor.py

```diff
@@ -51,8 +51,12 @@ async def paginate_aggregate(
     )
 
     data = (await cursor.to_list(length=None))[0]
-    total = data["metadata"][0]["total"]
+
     items = data["data"]
+    try:
+        total = data["metadata"][0]["total"]
+    except IndexError:
+        total = 0
```

Comment:
What do you think about using the get method instead of a try-except block to access the values?

ex:

```
items = data.get("data", [])
metadata = data.get("metadata", [{}])
total = metadata[0].get("total", 0)
```
or this: 

```
total = data.get("metadata", [{}])[0].get("total", 0)
```

Reply:
@AFU92 Your code will raise the IndexError if `metadata` is an empty list, which is actually the bug I've tried to fix.

`data.get("metadata", [{}])` would make sense if the key `metadata` doesn't exists, but it always does. If the key exists and `metadata` is an empty list, `metadata[0]` would raise the exception.

An option could be something like `total = metadata[0].get("total", 0) if metadata else 0` or also:

```
total = 0
if metadata:
    total = metadata[0].get("total", 0)
```

### Performance comparision
I've compared the performance of the three scenarios, and there are two possible situations.

1. metadata is a list with values, so we need to get the total key of the first element inside the list.
2. metadata is an empty list.

The faster option when the metadata is an empty list is evaluating it using the `if` statement.
The faster option when the metadata contains resulsts is the try/except block.
In both options, the slowest one is using the pattern-matching

Raising an exception is more expensive than making an `if` statement, but the try/except block is better when the exception is not raised (because we do an unnecessary `if` some times).

So, it's depends, all options will work.

This is a pagination library, so I would expect that metadata contains data more ofthen than an empty list.

Having that in mind, I would say that the try/except block will have a better performance in general.

What do you think @uriyyo @AFU92 ?